sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,356 @@
+        ((h/4 - lB/2)*omega_J1(t) + (h/4 + lB/2)*omega_J1(t))*rod_frame.x
+        >>> rod.frame.ang_vel_in(C.frame)
+        omega_J1(t)*C_frame.y
+        >>> plate.frame.ang_vel_in(C.frame)
+        omega_J2(t)*rod_frame.z + omega_J1(t)*C_frame.y
+        >>> rod.frame.dcm(C.frame)
+        Matrix([
+        [cos(theta_J1(t)), 0, -sin(theta_J1(t))],
+        [               0, 1,                 0],
+        [sin(theta_J1(t)), 0,  cos(theta_J1(t))]])
+        >>> plate.frame.dcm(C.frame)
+        Matrix([
+        [ cos(theta_J1(t))*cos(theta_J2(t)), sin(theta_J2(t)), -sin(theta_J1(t))*cos(theta_J2(t))],
+        [-sin(theta_J2(t))*cos(theta_J1(t)), cos(theta_J2(t)),  sin(theta_J1(t))*sin(theta_J2(t))],
+        [                  sin(theta_J1(t)),                0,                   cos(theta_J1(t))]])
+
+    Notes
+    ======
+
+    All dynamicsymbols used in `Joints` are `real ` by default. It is adviced to use
+    `real=True` assumption when passing dynamicsymbols to `Joints` else it may give
+    incorrect result in some cases.
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis=None, child_axis=None):
+
+        super().__init__(name, parent, child, coordinates, speeds, parent_joint_pos, child_joint_pos,
+            parent_axis, child_axis)
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            theta = dynamicsymbols('theta' + '_' + self._name, real=True)
+            coordinate = theta
+        coordinates.append(coordinate)
+        return coordinates
+
+    def _generate_speeds(self, speed):
+        speeds = []
+        if speed is None:
+            omega = dynamicsymbols('omega' + '_' + self._name, real=True)
+            speed = omega
+        speeds.append(speed)
+        return speeds
+
+    def _orient_frames(self):
+        self._child.frame.orient_axis(self._parent.frame, self._parent_axis, 0)
+        angle, axis = self._align_axis(self._parent_axis, self._child_axis)
+        if axis != Vector(0):
+            I = ReferenceFrame('I')
+            I.orient_axis(self._child.frame, self._child_axis, 0)
+            I.orient_axis(self._parent.frame, axis, angle)
+            self._child.frame.orient_axis(I, self._parent_axis, self._coordinates[0])
+        else:
+            self._child.frame.orient_axis(self._parent.frame, self._parent_axis, self._coordinates[0])
+
+    def _set_angular_velocity(self):
+        self._child.frame.set_ang_vel(self._parent.frame, self._speeds[0] * self._parent_axis.normalize())
+
+    def _set_linear_velocity(self):
+        self._parent_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_pos(self._parent_joint, 0)
+        self._child.masscenter.v2pt_theory(self._parent.masscenter, self._parent.frame, self._child.frame)
```

Comment:
Does this also set the opposite velocity, parent mass center relative to the child mass center?